### PR TITLE
[doc] custom output formatter schema

### PR DESCRIPTION
**Pull request by sindhuvahinis**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Review by lanking520 — APPROVED** [review]

File: serving/docs/lmi/user_guides/output_formatter_schema.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    TextGenerationOutput "1" --> "1..*" Sequence
+    Sequence "1" --> "1..*" Token
+    RequestInput <|-- TextInput
+    class RequestOutput{
+        +int request_id
+        +bool finished
+        +RequestInput input
+    }
+    class TextGenerationOutput{
+        +map sequences
```
Comment (siddvenk) [inline serving/docs/lmi/user_guides/output_formatter_schema.md:31]: what is the mapping?

File: serving/docs/lmi/user_guides/output_formatter_schema.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        +list[Token] tokens
+        +float cumulative_log_prob
+        +string finish_reason
+        +has_next_token()
+        +get_next_token()
+    }
+    
+    class RequestInput{
+        +int request_id
+        +dict parameters
```
Comment (siddvenk) [inline serving/docs/lmi/user_guides/output_formatter_schema.md:45]: we used map previously, and dict here. Let's keep it consistent (or maybe i'm not understanding the difference between the two).

If they are the same, should we say what the mapping is? (from what to what)

File: serving/docs/lmi/user_guides/output_formatter_schema.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  - has_next_token() and get_next_token() methods function like an iterator. In iterative generation, each step produces a single token.
+  - get_next_token() advances the iterator to the next token and returns a Token instance along with details indicating whether it is the first token (first_token) and whether it is the last token (last_token).
+
+## Example
+Here is an example of a custom output formatter:
+```python
+from djl_python.output_formatter import TextGenerationOutput
+import json
+
+def custom_output_formatter(request_output: TextGenerationOutput):
```
Comment (siddvenk) [inline serving/docs/lmi/user_guides/output_formatter_schema.md:80]: nit: let's add the return type here `-> str` to keep it consistent with the method signature we explain at the top

File: serving/docs/lmi/user_guides/output_formatter_schema.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    Args:
+        request_output (TextGenerationOutput): The request output
+
+    Returns:
+        (str): Response string
+
+    """
+    best_sequence = request_output.sequences[request_output.best_sequence_index]
+    next_token, first_token, last_token = best_sequence.get_next_token()
```
Comment (siddvenk) [inline serving/docs/lmi/user_guides/output_formatter_schema.md:92]: out of scope here, but it seems confusing that `get_next_token` returns 3 token values? i'll connect with you later to understand this more

Comment (sindhuvahinis) [reply serving/docs/lmi/user_guides/output_formatter_schema.md:92]: yes it might be more clear if its `next_token, is_first_token, is_last_token`

**Pull request merged**